### PR TITLE
feature/improving-class-constructors

### PR DESCRIPTION
**Pull request by BAILOOL**
Minor fixes to prevent class inheritance and remove (redundant) default class constructors.

### CODE REVIEW

File: src/Camera.cpp [hunk 1]
```diff
@@ -8,7 +8,10 @@
 #include "OptimizationCeres.h"
 #include "logger.h"
 
-Camera::Camera() {}
+Camera::Camera(const int cam_idx, const int distortion_model) {
+  cam_idx_ = cam_idx;
+  distortion_model_ = distortion_model;
+}
```
Comment (zzangjinsun) [inline src/Camera.cpp:14]: Camera::Camera(const int cam_idx, const int distortion_model) : cam_idx_(cam_idx), distortion_model_(distortion_model) {}; 
 
can be an option

File: src/Object3D.cpp [hunk 2]
```diff
@@ -30,9 +28,8 @@ void Object3D::insertNewObject(std::shared_ptr<Object3DObs> new_object) {
  * @param obj_id object id
  * @param color color of the 3D object
  */
-void Object3D::initializeObject3D(const int nb_boards, const int ref_board_id,
-                                  const int obj_id,
-                                  const std::vector<double> color) {
+Object3D::Object3D(const int nb_boards, const int ref_board_id,
```
Comment (zzangjinsun) [inline src/Object3D.cpp:31]: same here

File: src/Object3DObs.cpp [hunk 3]
```diff
@@ -9,16 +9,14 @@
...
-
 /**
  * @brief initialize the object in the observation (what board is observed)
  *
  * @param new_obj_obs object to be added
  * @param object_idx object index
  */
-void Object3DObs::initializeObject(std::shared_ptr<Object3D> obj_obs,
-                                   const int object_idx) {
+Object3DObs::Object3DObs(std::shared_ptr<Object3D> obj_obs,
```
Comment (zzangjinsun) [inline src/Object3DObs.cpp:18]: same here

**Review by zzangjinsun — APPROVED** [review]

File: src/Camera.cpp (hunk 1 above)
Comment (BAILOOL) [reply src/Camera.cpp]: Incorporated this feedback in the latest commit

File: src/CameraGroup.cpp [hunk 4]
```diff
@@ -9,18 +9,15 @@
...
+ * @brief Initialize the camera group object
  *
  * Set the index of the reference camera and the index of the camera group.
  *
  * @param id_ref_cam index of the reference camera
  * @param cam_group_idx index of the camera group
  */
-void CameraGroup::initializeCameraGroup(const int id_ref_cam,
-                                        const int cam_group_idx) {
+CameraGroup::CameraGroup(const int id_ref_cam, const int cam_group_idx) {
```
[the remark this answers is not in the document]
Comment (BAILOOL) [reply src/CameraGroup.cpp]: Incorporated this feedback in the latest commit

File: src/Object3D.cpp (hunk 2 above)
Comment (BAILOOL) [reply src/Object3D.cpp:31]: Incorporated this feedback in the latest commit

File: src/Object3DObs.cpp (hunk 3 above)
Comment (BAILOOL) [reply src/Object3DObs.cpp:18]: Incorporated this feedback in the latest commit

**Review by rameau-fr — APPROVED** [review]

**Pull request merged**